stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/common/avatar/ChannelAvatar.kt

```diff
@@ -0,0 +1,118 @@
+package io.getstream.chat.android.compose.ui.common.avatar
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.RectangleShape
+import coil.compose.rememberImagePainter
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.models.image
+import io.getstream.chat.android.client.models.initials
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+
+/**
+ * Represents the [Channel] avatar that's shown when browsing channels or when you open the Messages screen.
+ *
+ * Based on the state of the [Channel] and the number of members, it shows different types of images.
+ *
+ * @param channel - The channel whose data we need to show.
+ * @param currentUser - The current user, used to determine avatar data.
+ * @param modifier - Modifier for styling.
+ * @param contentDescription - The description to use for the avatar.
+ * */
+@Composable
+public fun ChannelAvatar(
+    channel: Channel,
+    currentUser: User?,
+    modifier: Modifier = Modifier,
+    contentDescription: String? = null,
+) {
+    val members = channel.members
+    val memberCount = members.size
+
+    when {
+        /**
+         * If the channel has an image we load that as a priority.
+         * */
+        channel.image.isNotEmpty() -> {
+            val painter = rememberImagePainter(data = channel.image)
+
+            Avatar(
+                modifier = modifier,
+                painter = painter,
+                contentDescription = contentDescription
+            )
+        }
+        /**
+         * If the channel has just one member (current user) we show our initials.
+         * */
+        memberCount == 1 -> {
+            val channelInitials = channel.initials
+
+            InitialsAvatar(initials = channelInitials, modifier)
+        }
+        /**
+         * If the channel has two members - direct message with another person - we show their image or initials.
+         * */
+        memberCount == 2 -> {
+            val user = members.first { it.user.id != currentUser?.id }.user
+
+            UserAvatar(
+                modifier = modifier,
+                user = user,
+                contentDescription = user.name
+            )
+        }
+        /**
+         * If the channel has more than two members - group - we load a matrix of their images or initials.
+         * */
+        else -> {
```

Comment:
Does the logic here work nicely for 3 avatars? 🤔

Reply:
As per the before/after images, it should work the same as for attachments - we show 2 avatars on the left side, split screen, and the third avatar on the right (full height).